biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -292,24 +292,52 @@ public Workspace(File workspaceDir) throws Exception {
 	}
 
 	public Workspace(File workspaceDir, String bndDir) throws Exception {
-		super(getDefaults());
+		super(new Processor(getDefaults()));
 		this.maven = new Maven(Processor.getExecutor(), this);
 		this.layout = WorkspaceLayout.BND;
 		workspaceDir = workspaceDir.getAbsoluteFile();
 		setBase(workspaceDir); // setBase before call to setFileSystem
 		addBasicPlugin(new LoggingProgressPlugin());
 		setFileSystem(workspaceDir, bndDir);
+
+		// we must process version defaults after the
+		// normal properties are read
+
+		fixupVersionDefaults(this.getParent()
+			.getProperties());
+
 		projects = new ProjectTracker(this);
 	}
 
 	private Workspace(WorkspaceLayout layout) throws Exception {
-		super(getDefaults());
+		super(new Processor(getDefaults()));
 		this.maven = new Maven(Processor.getExecutor(), this);
 		this.layout = layout;
 		setBuildDir(IO.getFile(BND_DEFAULT_WS, CNFDIR));
+		// we must process version defaults after the
+		// normal properties are read
```

Comment:
Nope. I am using an intermediate processor. 

           defaults <- version defaults <- workspace <- project
                                         ^^^

The version defaults is a new intermediate. Only this processor is changed. If you set a property in the workspace, it will be found before the version defaults and/or overall defaults. 

We cannot change the overall defaults since this would interfere with different workspaces that would pick different version defaults.